File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/state/orderstate/OrderReceiveState.java
// OrderReceiveState.java

package com.bigboxcode.designpattern.state.orderstate;

public class OrderReceiveState extends OrderState {
    public OrderReceiveState(OrderContext context) {
        super(context);
    }

    @Override
    public void process() {
        // Write code to process the order
        System.out.println("Order received");

        context.setState(null);
    }
}

File: design-patterns/php/state/order/OrderReceiveState.php
<?php
// OrderReceiveState.php

namespace BigBoxCode\DesignPattern\State\Order;


class OrderReceiveState extends OrderState {
    public function __construct(OrderContext $context) {
        parent::__construct($context);
    }

    public function process(): void {
        // Write code to process the order
        echo "Order received\n";

        $this->context->setState(null);
    }
}